contracts/verifiers/Verifier.sol

```diff
@@ -192,41 +190,10 @@ abstract contract Verifier is IVerifier, ContextUpgradeable {
     function setRequests(IVerifier.Request[] calldata requests) public {
         VerifierStorage storage s = _getVerifierStorage();
 
-        uint256 newGroupsCount = 0;
-        uint256[] memory newGroupsGroupID = new uint256[](requests.length);
-        uint256[] memory newGroupsRequestCount = new uint256[](requests.length);
-
         // 1. Check first that groupIds don't exist and keep the number of requests per group.
-        for (uint256 i = 0; i < requests.length; i++) {
-            uint256 groupID = requests[i]
-            .validator
-            .getRequestParams(requests[i].params)[
-                requests[i].validator.requestParamIndexOf("groupID")
-            ].value;
-
-            if (groupID != 0) {
-                if (groupIdExists(groupID)) {
-                    revert GroupIdAlreadyExists(groupID);
-                }
-
-                (bool exists, uint256 groupIDIndex) = _getGroupIDIndex(
-                    groupID,
-                    newGroupsGroupID,
-                    newGroupsCount
-                );
-
-                if (!exists) {
-                    newGroupsGroupID[newGroupsCount] = groupID;
-                    newGroupsRequestCount[newGroupsCount]++;
-                    newGroupsCount++;
-                } else {
-                    newGroupsRequestCount[groupIDIndex]++;
-                }
-            }
-        }
-
-        _checkGroupsRequestsCount(newGroupsGroupID, newGroupsRequestCount, newGroupsCount);
+        _checkGroupIdsAndRequestsPerGroup(requests);
 
+        bool userIDInRequests = false;
```

Comment:
Please check that all stand alone requests have `userID` in public inputs and check that all the groups have at least one request with `userID`

Reply:
Done